clients/privacy-center/components/consent/NoticeDrivenConsent.tsx

```diff
@@ -208,20 +208,24 @@ const NoticeDrivenConsent = () => {
       id: consentRequestId,
       body: payload,
     });
-    if ("error" in result) {
+    const isError = "error" in result;
+    if (isError || !result.data.preferences) {
+      let description = "No preferences returned";
+      if (isError) {
+        description = typeof result.error === "string" ? result.error : "";
+      }
       toast({
         title: "An error occurred while saving user consent preferences",
-        description:
-          typeof result.error === "string" ? result.error : undefined,
+        description,
         ...ErrorToastOptions,
       });
       return;
     }
 
     // 2. Save the cookie and window obj on success
     const noticeKeyMap = new Map<string, boolean>(
-      result.data.map((preference) => [
-        preference.privacy_notice_history.notice_key || "",
+      result.data.preferences.map((preference) => [
```

Comment:
adding `.preferences` here is the only real change—the API now nests the response under `preferences`